birdhouse/pavics-compose.sh

```diff
@@ -202,6 +202,11 @@ do
       docker exec ${postgres_id} /postgres-setup.sh
     fi
 
+    # because server.outputpath in wps.cfg do not create the dir
+    for bird in finch flyingpigeon raven; do
+      docker exec $bird mkdir -p /data/wpsoutputs/$bird
```

Comment:
Given that we all agree that this is a temporary hack, I would opt to put that code in a separate script from the main one. We already have a neat way of doing this with pre and post compose scripts. `pavics-compose.sh` could search for those scripts in the `./config` folder in addition to the extra conf directories. Then, all we have to do is placing the mkdir command in the bird conf directory as a pre compose script.

If you agree with me, we could then clean `pavics-compose.sh` further by moving the custom script section for geoserver and postgres to their own config folder. In the end the main compose script would be cleaner and each component would have the ability to run custom command in the context of pavics. Because I agree with @tlvu that adding the mdir command to the docker is also not the better option.

Reply:
On my end, I'm ok to wait for a PyWPS fix if that avoids throw-away code.

Reply:
> we could then clean `pavics-compose.sh` further by moving the custom script section for geoserver and postgres to their own config folder. In the end the main compose script would be cleaner and each component would have the ability to run custom command in the context of pavics.

@dbyrns Unfortunately geoserver and postgres are not in a "component" layout.  We have been adding new components using the new pluggable design but we have not migrate all the existing pieces to this pluggable design :(

> On my end, I'm ok to wait for a PyWPS fix if that avoids throw-away code.

@huard no big deal with throw-away code, it's just this section of `mkdir` that is throw-away.  Waiting for PyWPS means not only waiting for a release of PyWPS but also the integration into our birds then a new build of all our birds.

Reply:
> @dbyrns Unfortunately geoserver and postgres are not in a "component" layout. We have been adding new components using the new pluggable design but we have not migrate all the existing pieces to this pluggable design :(

I know, I'm not suggesting to make them proper components, but only to extend the pre/post compose loop to include the "built-in" components in the config directory. This way any existing services could include custom scripts.

This time I agree that we can wait for PyWPS, but we should keep that in mind if component related stuff are to be included in the `pavics-compose.sh`

Reply:
> extend the pre/post compose loop to include the "built-in" components in the config directory. This way any existing services could include custom scripts.

Yeah that's a quick win.  But I'd say also a potential throwaway code if we migrate all existing components to the new pluggable design.  Potential usecase for a complete pluggable design is not all organisations deploying PAVICS will want all the current components activated.  Maybe they just want Thredds and their birds, not our birds.

Anyways, back to this PR, it's a cheap throwaway and a quick win.  I'd rather take it now than wait.  I can make the pre/post for `configs/` as well since we are into cheap throwaway for quick win.  But I am fine if we prefer to wait.

Note the other issue https://github.com/bird-house/finch/issues/160 probably also need this same `mkdir` hack or the matching proper fix on PyWPS side.  So either we solve both issues now or we wait for both issues.

Reply:
@huard Are you okay to merge this quick hack for this fix?  Not sure how much time I'll need to debug and perform the proper fix in PyWPS.  All the old birds (hummingbird, ...) using the old "Buildout way" are basically doing this same hack, ie mkdir themselve outside of PyWPS.